mgmtsystem/azure.py

```diff
@@ -315,19 +341,50 @@ def copy_blob_image(self, template, vm_name, storage_account, template_blob, sto
                        template_blob=template_blob,
                        storage_blob=storage_blob))
 
-    def remove_blob_image(self, vm_vhd, storage_account, storage_blob):
+    def remove_blob_image(self, vm_vhd):
+        vhd_name = os.path.split(urlparse.urlparse(vm_vhd).path)[1]
         self.run_script(
             """
             Invoke-Command -scriptblock {{
             $sourceContext = New-AzureStorageContext -StorageAccountName \"{storage_account}\" `
                             -StorageAccountKey \"{storage_key}\"
-            $blobRemove = Remove-AzureStorageBlob -Blob \"{vm_vhd}\" `
+            $blobRemove = Remove-AzureStorageBlob -Blob \"{vhd_name}\" `
                         -Context $sourceContext -Container \"{storage_blob}\"
             }}
-            """.format(vm_vhd=vm_vhd,
-                       storage_account=storage_account,
+            """.format(vhd_name=vhd_name,
+                       storage_account=self.storage_account,
                        storage_key=self.storage_key,
-                       storage_blob=storage_blob))
+                       storage_blob=self.storage_blob), True)
+
+    def remove_diags_container(self):
+        self.run_script(
+            """
+            Invoke-Command -scriptblock {{
+            $storeContext = New-AzureStorageContext -StorageAccountName \"{storage_account}\" `
+                            -StorageAccountKey \"{storage_key}\"
+            $storageContainer = Get-AzureStorageContainer -Context $storeContext
+            foreach ($container in $storageContainer) {{
+                if ($container.name -like 'bootdiagnostics-test*'){{
+                    Remove-AzureStorageContainer -Name $container.name -Context $storeContext -Force
+                }}
+            }}
+            }}
+            """.format(storage_account=self.storage_account,
+                       storage_key=self.storage_key), True)
+
+    def cleanup_azure_system(self, hours_delay=10):
```

Comment:
This shouldn't be in wrapanapi. This should be a cleanup script in our integration_tests codebase. wrapanapi can be used by anyone, and the information in here is very specific to manageiq testing.